test/test_lang.cc

```diff
@@ -28,7 +28,12 @@
 
 using namespace lang;
 
+#ifdef ROOT_PATH
```

Comment:
We have been using ```TC_DIR``` across our build for this.
How about removing the case disjunction which is going to be always painful to maintain and got for something similar to ```CUDA_HOME```?
In other words:
1. ```const std::string expected_file_path = std::string(TC_DIR) + "src/lang/test_expected/";``` always,
2. ```TC_DIR``` is exported both in fbcode and CMakeLists (you can rename all occurrences to ```TC_ROOT_DIR``` in a separate commit if you prefer to have ```ROOT``` in the name),
3. have a ```#ifndef TC_ROOT_DIR``` trigger an error to ensure errors at compile time rather than runtime 

Also, let's take the opportunity to move all those expected test files in ```test/expected``` rather than keeping them polluting the source tree.